src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -650,7 +652,26 @@ public BulkResponse executeBulk(BulkRequest bulk) throws IOException {
 
     final String errorInfo = errors.isEmpty() ? result.getErrorMessage() : errors.toString();
 
-    return BulkResponse.failure(retriable, errorInfo);
+    Map<IndexableRecord, BulkResultItem> failedResponses = new HashMap<>();
+    List<BulkResultItem> items = result.getItems();
+    for (int i = 0; i < items.size() && i < ((JestBulkRequest) bulk).records().size() ; i++) {
+      BulkResultItem item = items.get(i);
+      IndexableRecord record = ((JestBulkRequest) bulk).records().get(i);
+      if (item.error != null && item.id.equals(record.key.id)) {
```

Comment:
I see in the constructor of `BulkResultItem` that `id` can be set to `null`. 
Are we risking NPE here?
Seems `record.key.id` can't be `null`. `Objects.equals` is an option that works with `null`